### PR TITLE
Remove immutables 

### LINKED ISSUE #7931: [flow] migrate ast reducer off of immutable
There are a couple of reasons to stop using immutable in our reducers

1. it can be slower
2. it is harder to flow type. 

I started migrating our ast reducer [here] and would love some help finishing it up. I think the main thing is making sure we fix up the flow types and get all of the tests passing.

[here]: https://github.com/devtools-html/debugger.html/compare/master...jasonLaster:flow-ast?expand=1

### PR DESCRIPTION
**Pull request by errietta**
Fixes #7931 

Here's the Pull Request Doc
https://firefox-dev.tools/debugger.html/docs/pull-requests.html

### Summary of Changes

* remove immutables

### Test Plan

Run `npm run test` and `npm run flow`

### CODE REVIEW

File: src/actions/breakpoints/index.js [hunk 1]
```diff
@@ -352,7 +350,7 @@ export function addBreakpointAtLine(line: number) {
     const state = getState();
     const source = getSelectedSource(state);
 
-    if (!source || isEmptyLineInSource(state, line, source.id)) {
```
Comment (jasonLaster) [inline src/actions/breakpoints/index.js]: lets keep empty lines. Even though it is not available now it will be nice to fix it up tomorrow.

Comment (errietta) [reply src/actions/breakpoints/index.js]: That made the tests fail though?

Comment (errietta) [reply src/actions/breakpoints/index.js]: Should be good now :)

File: src/actions/ast/setInScopeLines.js [hunk 2]
```diff
@@ -9,10 +9,10 @@ import { getSourceLineCount } from "../../utils/source";
 
 import { range, flatMap, uniq, without } from "lodash";
 
-import type { ASTLocation } from "../../types";
+import type { AstLocation } from "../../workers/parser";
 import type { ThunkArgs } from "../types";
 
-function getOutOfScopeLines(outOfScopeLocations: ASTLocation[]) {
+function getOutOfScopeLines(outOfScopeLocations: ?(AstLocation[])) {
```
Comment (darkwing) [inline src/actions/ast/setInScopeLines.js]: Why the `?` change here?

**Comment by darkwing** [comment]
Wow, this is really good @errietta !

File: src/actions/ast/setInScopeLines.js (hunk 2 above)
Comment (errietta) [reply src/actions/ast/setInScopeLines.js]: `getOutOfScopeLines` has a check if `ASTLocation` is falsey

**Comment by jasonLaster** [comment]
This looks great

**Comment by jasonLaster** [comment]
Thanks @errietta!

**Pull request merged**